### DISCUSSION TITLE
Support permanent (308) redirects for default locale

### DISCUSSION THREAD

**Discussion by jasperhartong** [amannn/next-intl#544] [Q&A]
If the middleware is configured to use `localePrefix: "as-needed"`, urls are redirect from `/:locale/slug` to `/slug` with a 307 Temporary Redirect. From an SEO perspective it might be better to allow these redirects to be performed with a 308 Permanent Redirect.

As far as I can tell the middleware doesn't provide a way to configure this behaviour?

**Comment by amannn** [comment]
As far as I know, a 307 should be fine: https://webmasters.stackexchange.com/a/123092

**Reply by jasperhartong** [reply]
Thanks for the swift answer, going to double check it with my SEO consultant.

**Reply by aleksijohansson** [reply]
I think it makes sense that you would want a 307 when using locale detection, but when it comes to the situation described in the original question, I think a 308 is definitely needed.

**Comment by mcianc** [comment]
Any chance to custominze the 3xx status code? My seo experts prefer to have 301 instead

**Comment by aleksijohansson** [comment]
I would really need this possibility to set the redirect status to 308 as well

**Reply by aleksijohansson** [reply]
Based on https://github.com/amannn/next-intl/issues/666 the suggested method is to "overwrite the response status code in certain cases if the next-intl middleware returns a redirect".

This is solution is requires you to create a completely new response, since the status of responses is immutable, which is not great.

**Reply by mahiraltinkaya** [reply]
> Based on #666 the suggested method is to "overwrite the response status code in certain cases if the next-intl middleware returns a redirect".
> 
> This is solution is requires you to create a completely new response, since the status of responses is immutable, which is not great.

Hey [aleksijohansson](https://github.com/aleksijohansson) ;

if you want to redirect as 308 you can add a basically code your nextjs middleware like below;


```
export default async function middleware(req: NextRequest) {
const nextUrl = req.nextUrl;
const pathname = nextUrl.pathname;

if (pathname === "/") {
  return NextResponse.redirect(`/${defaultLanguage}`, {
    status: 308,
  });
}
  return intlMiddleware(req);
```
